### DISCUSSION TITLE
rounded-x-* and rounded-y-* utilities

### DISCUSSION THREAD

**Discussion by PositiveUser** [tailwindlabs/tailwindcss#5959] [Ideas]
_@tailwindlabs_

Sometimes, I want to apply border radius to (top and bottom) or (left and right) sides of shape.

In this case, `rounded-x-*` and `rounded-y-*` utilities can help.

**For example**

`rounded-x-lg` will apply `rounded-l-lg` and `rounded-r-lg`

Similarly, `rounded-y-lg` will apply `rounded-t-lg` and `rounded-b-lg`

---
Thank you very much

**Comment by NovusOpus** [comment]
But the `rounded-x`  or `rounded-y` will end up doing the same as `rounded`. Just think about it :)

**Reply by PositiveUser** [reply]
No. Top and bottom have the same border radius. Left and right have the same border radius. But, the border radius of x axis is different from border radius of y axis.

top == bottom
left == right

x_border_radius != y_border_radius

**Reply by adamwathan** [reply]
Can you create an example in vanilla CSS? The problem is radius applies to corners, not sides.

`rounded-x` would set `border-top-left-radius`, `border-bottom-left-radius`, `border-top-right-radius`, and `border-bottom-right-radius`.

`rounded-y` would also set `border-top-left-radius`, `border-bottom-left-radius`, `border-top-right-radius`, and `border-bottom-right-radius`.

So `rounded-x` would target these corners:

![image](https://user-images.githubusercontent.com/4323180/140036914-941596e9-6dd0-486d-a030-f3c7ad96da85.png)

...and `rounded-y` would target these corners:

![image](https://user-images.githubusercontent.com/4323180/140036851-f248f907-df94-4650-b74f-f5c06b5b94c8.png)

The problem is even though I've drawn a different oval, they are still the same corners, both utilities would be setting all four corners.

You can't set the radius for the top without also setting the radius for the left and right sides unless I'm misunderstanding 🤔

**Reply by adamwathan** [reply]
Do you mean like this sort of thing? Not sure how to make this work as an actual API though:

https://play.tailwindcss.com/BPcsUzN6cS?file=css

**Reply by NovusOpus** [reply]
@PositiveUser As you can see in this [TailwindPlay](https://play.tailwindcss.com/l3HI8R7Nh8?file=css).

They end up looking the same:

Every color is a size, green is the default, blue is `md` and indigo is `lg`, 

First column is the default tailwind utility `rounded`, second column is your suggestion for `rounded-x` 

And the last column is your suggestion to `rounded-y`

**Edit: Typo, replaced breakpoint for size**

**Comment by PositiveUser** [comment]
Ok. Thank you very much. Delete this if you can delete this && you want to delete this
